### DISCUSSION TITLE
manipulate url string for api fetch / database query

### DISCUSSION THREAD

**Discussion by marvinRoellecke** [neuefische/questions#446] [Web] [answered]
Hi,
I would like to filter my card list using database queries. 

The idea is to trigger a new fetch with different url strings depending on selected filter (checkboxes) e.g.:
  `/api/locations/?sport=basketball`
or if multiple filters selected
`/api/locations/?sport=basketball&sport=tennis&city=munich`

if a filter is unchecked the query string should be removed

I thought the Web API URLSearchParams() would be the right, adding and removing queries with `URLSearchParams.append()` and `URLSearchParams.delete()`
But unfortunately it doesn't work as expected. (Maybe because of React??)
I am not able to update the url when click on a checkbox

The query itself works, tested with Postman

**I "only" struggle with manipulating the url string.**

**Happy to get a hint where I can find a solution. Many thanks**

**Comment by pixelass** [comment]
Do you have a code example? Have you tried to isolate the problem in a codepen or codesandbox?
I prepared a small example for you (look at the console)

https://codesandbox.io/s/wonderful-cache-7z05pf?file=/src/index.js



FYI: definitely not caused by react, since React does not modify any global Vanilla JavaScript methods.

**Reply by marvinRoellecke** [reply]
Thanks for quick response!

Ah I think your example led me into the right direction. 
I thought .append modifies the url variable directly, but I need to use the .toString() and add this to the api url.
I will implement this to my code and post the result here 

Thank you.

**Comment by marvinRoellecke** [comment, marked as answer]
getting closer...

Now it's possible to add multiple url strings to the url. And it works perfectly fine if I just log out the searchURL.

But when triggering the fetch the searchURL only contains the last selected search parameter.
Now I'm stuck at this stage

```
const params = new URL("http://localhost:3000/api/locations/?");

async function handleFilter(event, category) {

params.searchParams.append(category, event.target.value);
const searchURL = params.searchParams.toString();

 try {
      const response = await fetch("/api/locations/?" + searchURL);
      const data = await response.json();
      setLocations(data);
    } catch (error) {
      console.error(error);
    }
}
```

**Reply by marvinRoellecke** [reply]
use useStare instead of normal const for params seems to work.
setter function not used

```
 const [params, setParams] = useState(
    new URL("http://localhost:3000/api/locations/?")
  );
```
